### ISSUE TITLE
Disambiguation between errors and warnings when loading models

### ISSUE THREAD

**Issue by stillonearth** [aaronlaw1215/MuJoCo-Rust#9]
In some cases MuJoCo will throw an error such

```Error loading model: Degenerate triangle found in mesh `trunk`, its orientation will not be checked.```

I'm getting this from loading https://github.com/deepmind/mujoco_menagerie/blob/main/unitree_a1/a1.xml

This will fail in 

```rust
fn from_xml_helper(
    model_ptr: *mut mjModel,
    err_buf: Vec<u8>,
) -> Result<Model, String> {
    debug_assert_ne!(err_buf.len(), 0);
    let err_str = crate::helpers::convert_err_buf(err_buf);

    if !err_str.is_empty() { // Here
        return Err(err_str);
    }
    if model_ptr.is_null() {
        unreachable!(
            "It shouldn't be possible to get a null pointer from mujoco \
                without an error message!"
        );
    }
    Ok(Model { ptr: model_ptr })
}
```

Currently there's no way to separate errors and warnings.

**Comment by stillonearth** [comment]
Okay it seems that's a mujoco bug, that's fixed recently
https://github.com/deepmind/mujoco/discussions/591

**Comment by stillonearth** [comment]
Fixed with MuJoCo 2.3.1
